Source/Shared/StorageLegacy/DiskStorage.swift

```diff
@@ -0,0 +1,278 @@
+import Foundation
+import SwiftHash
+
+/**
+ File-based cache storage.
+ */
+final class DiskStorage: StorageAware {
```

Comment:
Why do we need this?

Reply:
I move conflicted files into `Legacy folder`, they will be removed later